test/base-tlsutility.cpp

```diff
@@ -35,4 +84,64 @@ BOOST_AUTO_TEST_CASE(sha1)
 	}
 }
 
+BOOST_AUTO_TEST_CASE(iscertuptodate_leaf_ok)
+{
+	BOOST_CHECK(IsCertUptodate(MakeCert("Icinga CA", GenKeypair(), "example.com", GenKeypair(), [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
+		time_t epoch = 0;
+		BOOST_REQUIRE(X509_time_adj(notBefore, l_2017, &epoch));
+		BOOST_REQUIRE(X509_gmtime_adj(notAfter, RENEW_THRESHOLD + 60 * 60));
+	})));
+}
+
+BOOST_AUTO_TEST_CASE(iscertuptodate_leaf_expiring)
+{
+	BOOST_CHECK(!IsCertUptodate(MakeCert("Icinga CA", GenKeypair(), "example.com", GenKeypair(), [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
+		time_t epoch = 0;
+		BOOST_REQUIRE(X509_time_adj(notBefore, l_2017, &epoch));
+		BOOST_REQUIRE(X509_gmtime_adj(notAfter, RENEW_THRESHOLD - 60 * 60));
+	})));
+}
```

Comment:
So these certificates use `l_2016` / 2016-11-24 as a fixed start but `now + RENEW_THRESHOLD ± 1h` as end? Why not just use now as the beginning? Would be closer to the actual certificates (and even make the code a bit simpler).

Reply:
Sure. But. There three tests:

1. Everything is OK, but as close to not OK as possible, everywhere
2. Just like (1), but slightly violates start of validation
3. Just like (1), but slightly violates end of validation

I've intentionally made them easy to compare like: oh, just a few hours more right here and boom!